### ISSUE TITLE
Memory Leak in BarChartView Data 

### ISSUE THREAD

**Issue by Atchu** [danielgindi/Charts#2416]
When I checked Memory Leaks in Instruments always showed BarchartView.data=data as memory leak

 ```
BarChartData *data1=nil;
    if(dataSets && dataSets.count>0)
    {
    data1 = [[BarChartData alloc] initWithDataSets:dataSets];
    [data1 setValueFont:[UIFont fontWithName:@"HelveticaNeue-Light" size:10.f]];
    }
    self.barChartView.data = data1; //memory Leak
```
any suggestions to solve the memory leak here

**Comment by liuxuan30** [comment]
Have you checked the retain count at last?

Last year we met some cases saying memory leak, but I asked some apple engineers, they said it's instruments bug. This code has not changed a long time, I'm not sure if it's really a leak.

Please post screenshots

**Comment by Atchu** [comment]
Thanks for Reply I just Attaching screen shot below 
<img width="965" alt="screen shot 2017-05-08 at 9 27 00 am" src="https://cloud.githubusercontent.com/assets/5966039/25799905/a8d63638-3404-11e7-8c01-c68b7500f6bc.png">

**Comment by liuxuan30** [comment]
No, I want instruments leak report, not the code. Actually, I can't find instruments complains what you post on my side.

**Comment by liuxuan30** [comment]
closed, dup to #2425
